arch/aarch64/mcount-support.c

```diff
@@ -110,10 +110,44 @@ int mcount_get_register_arg(struct mcount_arg_context *ctx,
 	return 0;
 }
 
+struct mem_region {
+	struct rb_node          node;
+	unsigned long           start;
+	unsigned long           end;
+};
+
+bool check_arg_mem_region(struct mcount_mem_regions *regions,
+			  unsigned long addr)
+{
+	struct rb_root *root = &regions->root;
+	struct rb_node *parent = NULL;
+	struct rb_node **p = &root->rb_node;
+	struct mem_region *iter;
+
+	if (regions->heap <= addr && addr < regions->brk)
+		return true;
+
+	while (*p) {
+		parent = *p;
+		iter = rb_entry(parent, struct mem_region, node);
+
+		if (iter->start <= addr && addr < iter->end)
+			return true;
+
+		if (iter->start > addr)
+			p = &parent->rb_left;
+		else
+			p = &parent->rb_right;
+	}
+
+	return false;
+}
```

Comment:
Please just use existing `check_mem_region()` instead of duplicating the code for each arch.  You'll need to make it a global function (by removing `static` keyword) first.

Reply:
The `check_mem_region` function is located between `#ifdef DISABLE_MCOUNT_FILTER` and `#endif` so that it will be compiled only when the `DISABLE_MCOUNT_FILTER` flag is set.

To use this function as a global function, it must be changed from conditional compilation to unconditional compilation, as well as removing the static keyword.

In addition, some of the functions and structures associated with this function are set to conditional compilation, which must be changed to unconditional compilation.

Is it okay to change like this?